sql/ir_generator.go

```diff
@@ -367,6 +400,74 @@ func parseEmbeddingColumn(el *exprlist) (*codegen.EmbeddingColumn, error) {
 		Initializer:    initializer}, nil
 }
 
+func parseFieldMeta(el *exprlist) (*codegen.FieldMeta, error) {
+	help := "DENSE|SPARSE(col_name, SHAPE, DELIMITER[, DTYPE])"
+	if len(*el) < 4 {
+		return nil, fmt.Errorf("bad FieldMeta format: %s, should be like: %s", *el, help)
+	}
+	call, err := expression2string((*el)[0])
+	if err != nil {
+		return nil, fmt.Errorf("bad FieldMeta format: %v, should be like: %s", err, help)
+	}
+	var isSparse bool
+	if strings.ToUpper(call) == "DENSE" {
+		isSparse = false
+	} else if strings.ToUpper(call) == "SPARSE" {
+		isSparse = true
+	} else {
+		return nil, fmt.Errorf("bad FieldMeta: %s, should be like: %s", call, help)
+	}
+
+	name, err := expression2string((*el)[1])
+	if err != nil {
+		return nil, fmt.Errorf("bad FieldMeta name: %s, err: %s", (*el)[1], err)
+	}
+	var shape []int
+	intShape, err := strconv.Atoi((*el)[2].val)
+	if err != nil {
+		strShape, err := expression2string((*el)[2])
+		if err != nil {
+			return nil, fmt.Errorf("bad FieldMeta shape: %s, err: %s", (*el)[2].val, err)
+		}
+		if strShape != "none" {
+			return nil, fmt.Errorf("bad FieldMeta shape: %s, err: %s", (*el)[2].val, err)
+		}
+	} else {
+		shape = append(shape, intShape)
+	}
+	unresolvedDelimiter, err := expression2string((*el)[3])
+	if err != nil {
+		return nil, fmt.Errorf("bad FieldMeta delimiter: %s, err: %s", (*el)[1], err)
+	}
+
+	delimiter, err := resolveDelimiter(unresolvedDelimiter)
+	if err != nil {
+		return nil, err
+	}
+
+	dtype := codegen.Float
+	if isSparse {
+		dtype = codegen.Int
+	}
+	if len(*el) >= 5 {
+		dtypeStr, err := expression2string((*el)[4])
+		if err != nil {
+			return nil, err
+		}
+		if dtypeStr == "float" {
+			dtype = codegen.Float
+		} else if dtypeStr == "int" {
+			dtype = codegen.Int
+		}
```

Comment:
Maybe add handling for unrecognized data types
```go
else {
    return nil, fmt.Errorf("bad FieldMeta data type ...")
}
```

Reply:
Done